modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -540,7 +608,11 @@ namespace cv {
                 CV_Assert(net->width > 0 && net->height > 0 && net->channels > 0);
 
                 int current_channels = net->channels;
+                int current_width = net->width;
+                int current_height = net->height;
+                int current_total = current_height * current_width * current_channels;
```

Comment:
Do we really need to track `current_total`? We can  just calculate it once for connected layer.

Reply:
@dkurt , Are you asking not to use extra space i.e. out_total_vec ?

https://github.com/opencv/opencv/blob/b9e10746e810d4128521dac48431e771c075e2bf/modules/dnn/src/darknet/darknet_io.hpp#L98

Then we will have to store output size in `out_channel_vec` of previous layer.
```
 else if (layer_type == "connected")
 {
    int output = getParam<int>(layer_params, "output", 1);
    bool batch_normalize = getParam<int>(layer_params, "batch_normalize", 0) == 1;

    CV_Assert(output > 0);

    setParams.setFullyConnected(output, batch_normalize);

    if(layers_counter > 0)
    {
       if(current_width)
           net->out_channels_vec[layers_counter-1] = current_channels * current_width * current_height;
       else
           net->out_channels_vec[layers_counter-1] = current_channels;
    }
    current_channels = output;
    current_width = 0;
    current_height = 0;
}
```
No need to store `current_total`
This would work fine too but would be little ambiguous to store it in out_channel_vec.